pkg/agent/multicluster/mc_route_controller.go

```diff
@@ -212,20 +242,215 @@ func (c *MCDefaultRouteController) processNextWorkItem() bool {
+					if deleteErr := c.cleanUpWireGuard(); deleteErr != nil {
+						klog.ErrorS(deleteErr, "Failed to clean up WireGuard stale resources")
+					}
+				}
+			}()
+			if initErr = c.initializeWireGuard(); initErr != nil {
+				klog.ErrorS(initErr, "Failed to initialize WireGuard interface")
+				return initErr
+			}
+
+			if initErr = c.updateWireGuardPeers(); initErr != nil {
+				klog.ErrorS(initErr, "Failed to update WireGuard peers")
+				return initErr
+			}
+			c.wireGuardInitialized = true
+		}
+	} else {
+		if !isMyselfGateway {
+			klog.V(4).InfoS("Gateway does not exist or this Node is not Gateway, skip reconciling WireGuard")
+			return nil
+		}
+
+		// If the WireGuard has not been initialized. Some necessary configuration may not
+		// be ready. In this situation, we need to requeue the ClusterInfoImport again.
+		if !c.wireGuardInitialized {
+			return fmt.Errorf("WireGuard has not been initialized, requeue ClusterInfoImport")
+		}
+
+		ciImport, err := c.ciImportLister.ClusterInfoImports(c.namespace).Get(key)
+		if err != nil {
+			return err
+		}
+		return c.updateSingleWireGuardPeer(ciImport)
+	}
+
+	return nil
+}
+
+// updateWireGuardPeers lists all existing ClusterInfoImports and tries to add them as WireGuard peer.
+func (c *MCDefaultRouteController) updateWireGuardPeers() error {
+	klog.InfoS("Adding existing ClusterInfoImport as WireGuard peers")
+	ciImports, err := c.ciImportLister.ClusterInfoImports(c.namespace).List(labels.Everything())
+	if err != nil {
+		return err
+	}
+	for _, ciImport := range ciImports {
+		if err := c.updateSingleWireGuardPeer(ciImport); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+// updateSingleWireGuardPeer tries to update a WireGuard peer with ClusterInfoImport. If updating successfully,
+// it will also create host route the peer.
+func (c *MCDefaultRouteController) updateSingleWireGuardPeer(ciImport *mcv1alpha1.ClusterInfoImport) error {
+	klog.InfoS("Updating WireGuard peer with ClusterInfoImport", "ClusterInfoImport", ciImport)
+	if ciImport.Spec.WireGuard == nil || ciImport.Spec.WireGuard.PublicKey == "" {
+		klog.InfoS("ClusterInfoImport's WireGuard field has not been initialized, skip it", "ClusterInfoImport", ciImport)
+		return nil
+	}
+	serviceIP, serviceCIDR, err := net.ParseCIDR(ciImport.Spec.ServiceCIDR)
+	if err != nil {
+		return err
+	}
+	remoteWireGuardIP := serviceIP.Mask(serviceCIDR.Mask)
+	_, remoteWireGuardNet, err := net.ParseCIDR(remoteWireGuardIP.String() + "/32")
+	if err != nil {
+		return err
+	}
+
+	gatewayIP := net.ParseIP(ciImport.Spec.GatewayInfos[0].GatewayIP)
+	allowedIPs := []*net.IPNet{remoteWireGuardNet}
+	if err := c.wireGuardClient.UpdatePeer(ciImport.Name, ciImport.Spec.WireGuard.PublicKey, gatewayIP, 0, allowedIPs); err != nil {
+		klog.ErrorS(err, "Failed to update WireGuard peer", "ClusterInfoImport", ciImport)
+		return err
+	}
+	klog.InfoS("Updated WireGuard peer with ClusterInfoImport", "ClusterInfoImport", ciImport)
```

Comment:
Do you need the whole `ciImport` in the log?

Reply:
change to `ciImport.Name`.